### PR TITLE
Release 1.5.1: CC to AKO migration

### PR DESCRIPTION
**Pull request by akshayhavile**

### CODE REVIEW

File: docs/cc_to_ako.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# Cloud connector to AKO migration
+## Overview
+Cloud connector(CC) is not supported from Avi Controller versrion 18.2.10 onwards. To migrate existing workload present in kubernetes cluster from cloud connector to AKO, following two features will be used.
```
Comment (monotosh-avi) [inline docs/cc_to_ako.md:3]: We would have to support cc with all 18.2.x version, including versions > 18.2.10, but not 20.x

Comment (akshayhavile) [reply docs/cc_to_ako.md]: Addressed. Thanks.

**Review by monotosh-avi — APPROVED** [review]

**Review by monotosh-avi — APPROVED** [review]

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

**Comment by akshayhavile** [comment]
build ako

File: docs/cc_to_ako.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+Label namespace(s) in kubernetes cluster with same `key:value` pair.
+
+![Alt text](images/k8-namespace-labels.png)
+
+As shown in above snippet, "red" namespace is labelled with "app: migrate". This will result in deleting virtual services of red namespace from CC Avi-controller and creating new virtual services for red namespace in AKO Avi-controller. There will be traffic disruption during migration.
+
+### 6. Internal testing
+
+When objects migrates to new AKO Avi controller, it acquires new VIP. With that new VIP, do internal traffic test for migrated applications.
```
Comment (sudswasavi) [inline docs/cc_to_ako.md:76]: This implies that VIP preservation is not guaranteed, maybe worth calling that out explicitly.


Consider changing the sentence to:

The newly created objects in the Avi Controller that AKO operates on may get new VIPs. Hence internal traffic tests need to be performed for this migrated applications.

File: docs/cc_to_ako.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+After successful testing, objects from remaining namespaces can be migrated by labelling namespace with valid label (as done in step 5).
```
Comment (sudswasavi) [inline docs/cc_to_ako.md:78]: Objects are migrated sequentially by labelling the namespace one by one.

File: docs/cc_to_ako.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+### 7. Client traffic redirection
+
+Once all ingresses, L4 services migrated to AKO Avi controller, client traffic can be redirected to new Vips by
```
Comment (sudswasavi) [inline docs/cc_to_ako.md:82]: Once all L4-L7 applications are migrated to AKO based Avi Controller...

File: docs/cc_to_ako.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+* either changing DNS server entry in corporate DNS server to point out to new DNS service from AKO based Vcenter Cloud
```
Comment (sudswasavi) [inline docs/cc_to_ako.md:83]: Do you need a bullet here? The sentence seems to start with a small letter.

Change the DNS server entry in corporate DNS server to point to the new DNS service on the AKO based Avi Controller.

File: docs/cc_to_ako.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+* or changing dns service IP address, of Vcenter cloud, to old CC dns VIP.
```
Comment (sudswasavi) [inline docs/cc_to_ako.md:84]: This sentence is a bit ambiguous, I couldn't understand what it implies.

**Review by sudswasavi — COMMENTED** [review]
When we are dealing with migration docs, it's a good idea to call out the caveats very clearly. You may also want to state clearly on which environment the migration is supported.

File: docs/cc_to_ako.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+*  Setup a new Vcenter cloud with IPAM.
+*  Setup a DNS Service
+*  Replicate AVI side objects referred by Ingresses/Services as part of AVI_PROXY annotations to new controller.
+
+### 2. Deploy AKO CRDs
+
+For each AVI_PROXY annotation present in an ingress, create corresponding AKO Http rule/Host rule/ AviInfrasetting rule.
+Details of these CRDs can be found out [here](crds/overview.md)
+
+If any AVI_PROXY annotation is not supported by these CRD's, then VS which is migrated to AKO will not have same features as that in CloudConnector.
```
[the remark this answers is not in the document]
Comment (akshayhavile) [reply docs/cc_to_ako.md:44]: List of Avi_proxy annotation supported is not upto date. So I didn't mention here.

**Comment by akshayhavile** [comment]
build ako

**Review by chauhanshubham — APPROVED** [review]

**Comment by chauhanshubham** [comment]
@akshayhavile this has conflicts, could you please rebase.

**Pull request merged**